driver-scala/src/it/scala/org/mongodb/scala/syncadapter/SyncMongoClient.scala

```diff
@@ -0,0 +1,95 @@
+package org.mongodb.scala.syncadapter
+
+import scala.collection.JavaConverters._
+import com.mongodb.ClientSessionOptions
+import com.mongodb.client.{ ClientSession, MongoClient => JMongoClient, MongoDatabase => JMongoDatabase }
+import org.bson.Document
+import org.bson.conversions.Bson
+import org.mongodb.scala.MongoClient
+import org.mongodb.scala.bson.DefaultHelper.DefaultsTo
+
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.reflect.ClassTag
+
+case class SyncMongoClient(wrapped: MongoClient) extends JMongoClient {
+
+  override def getDatabase(databaseName: String): JMongoDatabase =
+    new SyncMongoDatabase(wrapped.getDatabase(databaseName))
+
+  override def startSession: ClientSession =
+    SyncClientSession(Await.result(wrapped.startSession().head(), Duration(10, "second")), this)
+
+  override def startSession(options: ClientSessionOptions): ClientSession =
+    SyncClientSession(Await.result(wrapped.startSession(options).head(), Duration(10, "second")), this)
+
+  override def close(): Unit = wrapped.close()
+
+  override def listDatabaseNames = throw new UnsupportedOperationException
+
+  override def listDatabaseNames(clientSession: ClientSession) = throw new UnsupportedOperationException
+
+  override def listDatabases = new SyncListDatabasesIterable[Document](wrapped.listDatabases[Document]())
+
+  override def listDatabases(clientSession: ClientSession) = throw new UnsupportedOperationException
+
+  override def listDatabases[TResult](resultClass: Class[TResult]) =
+    new SyncListDatabasesIterable[TResult](
```

Comment:
This is the downside of the implicits used in Scala.  The compiler is not clever enough to determine the `DefaultsTo` and `ClassTag[TResult]` from resultClass and I can't override `listDatabases` to add the implicit helpers. So I have to manually provide them to the `wrapped.listDatabases` call.

Reply:
Ack (though I don't fully understand this)